instrumentor/controllers/common.go

```diff
@@ -0,0 +1,382 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Credits: https://github.com/keyval-dev/odigos
+*/
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"github.com/go-logr/logr"
+	apiV1 "github.com/logzio/kubernetes-instrumentor/api/v1alpha1"
+	"github.com/logzio/kubernetes-instrumentor/common/consts"
+	"github.com/logzio/kubernetes-instrumentor/instrumentor/patch"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"time"
+)
+
+const (
+	SkipAnnotation             = "logz.io/skip"
+	LogTypeAnnotation          = "logz.io/application_type"
+	TracesInstrumentAnnotation = "logz.io/traces_instrument"
+)
+
+func shouldSkip(annotations map[string]string, namespace string) bool {
+	for k, v := range annotations {
+		if k == SkipAnnotation && v == "true" {
```

Comment:
What happens if someone inserts "True" instead of "true"? Would we want it to be case-insensitive?